Nightly build failures

Couple builds broken this morning. It looks like it's due to recent PRs. @mperego do you want to take a look?

> Thanks @mcarlson801 .
> 
> My local build of omegah ([SCOREC/omega_h@94aa915](https://github.com/SCOREC/omega_h/commit/94aa915568abb01c169fc5b84bbbd12acede8935)) with the kokkos (4.5.00) cuda backend was successful.
> 
> Can someone post, or email me, the omegah relevant portions of the build log?

looks like it's a configure issue:
```
CMake Error at /home/projects/albany/nightlyCDashAlbanyBlake/build-intel/AlbBuildReleaseIntel/tpls/omega_h/omega_h-src/cmake/bob.cmake:346 (find_package):
  Could not find a package configuration file provided by "Kokkos" (requested
  version 3.7) with any of the following names:

    KokkosConfig.cmake
    kokkos-config.cmake

  Add the installation prefix of "Kokkos" to CMAKE_PREFIX_PATH or set
  "Kokkos_DIR" to a directory containing one of the above files.  If "Kokkos"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  /home/projects/albany/nightlyCDashAlbanyBlake/build-intel/AlbBuildReleaseIntel/tpls/omega_h/omega_h-src/CMakeLists.txt:63 (bob_add_dependency)
  ```

--- a/src/evaluators/state/PHAL_LoadStateField.hpp
+++ b/src/evaluators/state/PHAL_LoadStateField.hpp
@@ -37,13 +37,11 @@ private:
 
   using ExecutionSpace = typename PHX::Device::execution_space;
 
-  PHX::MDField<ScalarType> data;
+  PHX::MDField<ScalarType> field;
   std::string fieldName;
   std::string stateName;
 
   MDFieldMemoizer<Traits> memoizer;
-
-  MDFieldVectorRight<ScalarType> dataVec;
 };
 
 template<typename EvalT, typename Traits>
@@ -65,13 +63,11 @@ private:
 
   using ExecutionSpace = typename PHX::Device::execution_space;
 
-  PHX::MDField<ParamScalarT> data;
+  PHX::MDField<ParamScalarT> field;
   std::string fieldName;
   std::string stateName;
 
   MDFieldMemoizer<Traits> memoizer;
-
-  MDFieldVectorRight<ParamScalarT> dataVec;
 };
 
 // Shortcut names

--- a/src/evaluators/state/PHAL_LoadStateField_Def.hpp
+++ b/src/evaluators/state/PHAL_LoadStateField_Def.hpp
@@ -56,7 +56,7 @@ void LoadStateFieldBase<EvalT, Traits, ScalarType>::evaluateFields(typename Trai
   Kokkos::parallel_for(this->getName(),
                        Kokkos::MDRangePolicy<ExecutionSpace, Kokkos::Rank<3>>({0,0,0},{stateData.extent(0),stateData.extent(1),stateData.extent(2)}),
                        KOKKOS_CLASS_LAMBDA(const int i, const int j, const int k) {
-    field(i,j,k) = stateData(i,j,k);
+    field.access(i,j,k) = stateData.access(i,j,k);  //works also when rank is less than 3
   });
 }
 
@@ -102,7 +102,7 @@ void LoadStateField<EvalT, Traits>::evaluateFields(typename Traits::EvalData wor
   Kokkos::parallel_for(this->getName(),
                        Kokkos::MDRangePolicy<ExecutionSpace, Kokkos::Rank<3>>({0,0,0},{stateData.extent(0),stateData.extent(1),stateData.extent(2)}),
                        KOKKOS_CLASS_LAMBDA(const int i, const int j, const int k) {
-    field(i,j,k) = stateData(i,j,k);
+    field.access(i,j,k) = stateData.access(i,j,k); //works also when rank is less than 3
   });
 }
 

--- a/src/evaluators/state/PHAL_LoadStateField_Def.hpp
+++ b/src/evaluators/state/PHAL_LoadStateField_Def.hpp
@@ -22,10 +22,9 @@ LoadStateFieldBase(const Teuchos::ParameterList& p)
   fieldName =  p.get<std::string>("Field Name");
   stateName =  p.get<std::string>("State Name");
 
-  PHX::MDField<ScalarType> f(fieldName, p.get<Teuchos::RCP<PHX::DataLayout> >("State Field Layout") );
-  data = f;
+  field = PHX::MDField<ScalarType>(fieldName, p.get<Teuchos::RCP<PHX::DataLayout> >("State Field Layout") );
 
-  this->addEvaluatedField(data);
+  this->addEvaluatedField(field);
   this->setName("LoadStateField("+stateName+")"+PHX::print<EvalT>());
 }
 
@@ -34,7 +33,7 @@ template<typename EvalT, typename Traits, typename ScalarType>
 void LoadStateFieldBase<EvalT, Traits, ScalarType>::postRegistrationSetup(typename Traits::SetupData d,
                       PHX::FieldManager<Traits>& fm)
 {
-  this->utils.setFieldData(data,fm);
+  this->utils.setFieldData(field,fm);
 
   d.fill_field_dependencies(this->dependentFields(),this->evaluatedFields());
   if (d.memoizer_active()) memoizer.enable_memoizer();
@@ -51,15 +50,13 @@ void LoadStateFieldBase<EvalT, Traits, ScalarType>::evaluateFields(typename Trai
   //       whomever changed the data.
   const auto& stateToLoad = (*workset.stateArrayPtr)[stateName];
   auto stateData = stateToLoad.dev();
-  const int stateToLoad_size = stateToLoad.size();
 
-  MDFieldVectorRight<ScalarType> g(data);
-  dataVec = g;
+  ALBANY_ASSERT (stateData.rank() <= 3, "Current implementation supports only views with rank up to 3. If larger rank is needed modify code below");
 
   Kokkos::parallel_for(this->getName(),
-                       Kokkos::RangePolicy<ExecutionSpace>(0,data.size()),
-                       KOKKOS_CLASS_LAMBDA(const int i) {
-    dataVec[i] = (i < stateToLoad_size) ? stateData(i) : 0.0;
+                       Kokkos::MDRangePolicy<ExecutionSpace, Kokkos::Rank<3>>({0,0,0},{stateData.extent(0),stateData.extent(1),stateData.extent(2)}),
+                       KOKKOS_CLASS_LAMBDA(const int i, const int j, const int k) {
+    field(i,j,k) = stateData(i,j,k);
   });
 }
 
@@ -70,10 +67,10 @@ LoadStateField(const Teuchos::ParameterList& p)
   fieldName =  p.get<std::string>("Field Name");
   stateName =  p.get<std::string>("State Name");
 
-  PHX::MDField<ParamScalarT> f(fieldName, p.get<Teuchos::RCP<PHX::DataLayout> >("State Field Layout") );
-  data = f;
 
-  this->addEvaluatedField(data);
+  field = PHX::MDField<ParamScalarT>(fieldName, p.get<Teuchos::RCP<PHX::DataLayout> >("State Field Layout") );
+
+  this->addEvaluatedField(field);
   this->setName("Load State Field"+PHX::print<EvalT>());
 }
 
@@ -82,7 +79,7 @@ template<typename EvalT, typename Traits>
 void LoadStateField<EvalT, Traits>::postRegistrationSetup(typename Traits::SetupData d,
                       PHX::FieldManager<Traits>& fm)
 {
-  this->utils.setFieldData(data,fm);
+  this->utils.setFieldData(field,fm);
 
   d.fill_field_dependencies(this->dependentFields(),this->evaluatedFields());
   if (d.memoizer_active()) memoizer.enable_memoizer();
@@ -99,15 +96,13 @@ void LoadStateField<EvalT, Traits>::evaluateFields(typename Traits::EvalData wor
   //       whomever changed the data.
   const auto& stateToLoad = (*workset.stateArrayPtr)[stateName];
   auto stateData = stateToLoad.dev();
-  const int stateToLoad_size = stateToLoad.size();
 
-  MDFieldVectorRight<ParamScalarT> g(data);
-  dataVec = g;
+  ALBANY_ASSERT (stateData.rank() <= 3, "Current implementation supports only views with rank up to 3. If larger rank is needed modify code below");
 
   Kokkos::parallel_for(this->getName(),
-                       Kokkos::RangePolicy<ExecutionSpace>(0,data.size()),
-                       KOKKOS_CLASS_LAMBDA(const int i) {
-    dataVec[i] = (i < stateToLoad_size) ? stateData(i) : 0.0;
+                       Kokkos::MDRangePolicy<ExecutionSpace, Kokkos::Rank<3>>({0,0,0},{stateData.extent(0),stateData.extent(1),stateData.extent(2)}),
+                       KOKKOS_CLASS_LAMBDA(const int i, const int j, const int k) {
+    field(i,j,k) = stateData(i,j,k);
   });
 }
 

